tests/components/hassio/test_repairs.py

```diff
@@ -0,0 +1,435 @@
+    assert result
+
+    client = await hass_ws_client(hass)
+
+    await client.send_json({"id": 1, "type": "repairs/list_issues"})
+    msg = await client.receive_json()
+    assert msg["success"]
+    assert {
+        "breaks_in_ha_version": None,
+        "created": ANY,
+        "dismissed_version": None,
+        "domain": "hassio",
+        "ignored": False,
+        "is_fixable": False,
+        "issue_id": "unsupported_system_content_trust",
+        "issue_domain": None,
+        "learn_more_url": "https://www.home-assistant.io/more-info/unsupported/content_trust",
+        "severity": "warning",
+        "translation_key": "unsupported",
+        "translation_placeholders": {
+            "reason": "content_trust",
+        },
+    } in msg["result"]["issues"]
+    assert {
+        "breaks_in_ha_version": None,
+        "created": ANY,
+        "dismissed_version": None,
+        "domain": "hassio",
+        "ignored": False,
+        "is_fixable": False,
+        "issue_id": "unsupported_system_os",
+        "issue_domain": None,
+        "learn_more_url": "https://www.home-assistant.io/more-info/unsupported/os",
+        "severity": "warning",
+        "translation_key": "unsupported",
+        "translation_placeholders": {
+            "reason": "os",
+        },
+    } in msg["result"]["issues"]
+
+
+async def test_unhealthy_repairs_add_remove(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    hass_ws_client,
+):
+    """Test unhealthy repairs added and removed from dispatches."""
+    mock_resolution_info(aioclient_mock)
+
+    result = await async_setup_component(hass, "hassio", {})
+    assert result
+
+    client = await hass_ws_client(hass)
+
+    calls = async_mock_signal(hass, EVENT_SUPERVISOR_EVENT)
+    async_dispatcher_send(
+        hass,
+        EVENT_SUPERVISOR_EVENT,
+        {
+            ATTR_WS_EVENT: "health_changed",
+            "data": {
+                "healthy": False,
+                "unhealthy_reasons": ["docker"],
+            },
+        },
+    )
+    assert len(calls) == 1
+
+    await client.send_json({"id": 1, "type": "repairs/list_issues"})
+    msg = await client.receive_json()
+    assert msg["success"]
+    assert msg["result"] == {
+        "issues": [
+            {
+                "breaks_in_ha_version": None,
+                "created": ANY,
+                "dismissed_version": None,
+                "domain": "hassio",
+                "ignored": False,
+                "is_fixable": False,
+                "issue_id": "unhealthy_system_docker",
+                "issue_domain": None,
+                "learn_more_url": "https://www.home-assistant.io/more-info/unhealthy/docker",
+                "severity": "critical",
+                "translation_key": "unhealthy",
+                "translation_placeholders": {
+                    "reason": "docker",
+                },
+            },
+        ]
+    }
+
+    async_dispatcher_send(
```

Comment:
Can we send a websocket message with the client that indirectly dispatches a signal so we don't rely on that detail in the test?